EventFilter/SiPixelRawToDigi/src/PixelDataFormatter.cc

```diff
@@ -520,3 +406,76 @@ std::string PixelDataFormatter::print(const Word64& word) const {
   str << "word64:  " << reinterpret_cast<const bitset<64>&>(word);
   return str.str();
 }
+
+void PixelDataFormatter::unpackFEDErrors(PixelDataFormatter::Errors const& errors,
+                                         std::vector<int> const& tkerrorlist,
+                                         std::vector<int> const& usererrorlist,
+                                         edm::DetSetVector<SiPixelRawDataError>& errorcollection,
+                                         DetIdCollection& tkerror_detidcollection,
+                                         DetIdCollection& usererror_detidcollection,
+                                         edmNew::DetSetVector<PixelFEDChannel>& disabled_channelcollection,
+                                         DetErrors& nodeterrors) {
+  const uint32_t dummyDetId = 0xffffffff;
+  for (const auto& [errorDetId, rawErrorsVec] : errors) {
+    if (errorDetId == dummyDetId) {  // errors given dummy detId must be sorted by Fed
+      nodeterrors.insert(nodeterrors.end(), rawErrorsVec.begin(), rawErrorsVec.end());
+    } else {
+      edm::DetSet<SiPixelRawDataError>& errorDetSet = errorcollection.find_or_insert(errorDetId);
+      errorDetSet.data.insert(errorDetSet.data.end(), rawErrorsVec.begin(), rawErrorsVec.end());
+      // Fill detid of the detectors where there is error AND the error number is listed
+      // in the configurable error list in the job option cfi.
+      // Code needs to be here, because there can be a set of errors for each
+      // entry in the for loop over PixelDataFormatter::Errors
+
+      std::vector<PixelFEDChannel> disabledChannelsDetSet;
+
+      for (auto const& aPixelError : errorDetSet) {
+        // For the time being, we extend the error handling functionality with ErrorType 25
+        // In the future, we should sort out how the usage of tkerrorlist can be generalized
+        if (aPixelError.getType() == 25) {
```

Comment:
the meaning of FED Error 25 is different between phase-0 and phase-1, so the lines below should apply only for the phase-1 reconstruction.